### ISSUE TITLE
Equalize lengths of signals in SignalList

### ISSUE THREAD

**Issue by ajovanov95** [ajovanov95/probabilistic-spiking-neural-networks#2]
The goal of this task is to develop a new member function which will make sure that all signals have the same number of time steps.

This goal is to be achieved by use of padding with 0s at the end of the signals which are shorter.

Function signature:

```
// library/core/include/signal.h

void Signal::equalize_lengths();
```

Add unit tests as well.

**Comment by ajovanov95** [comment]
This feature will also require the addition of a new method to the `Signal` class with the following signature:

```c++
void Signal::pad(const uint32_t length);
```

**Comment by ajovanov95** [comment]
Resolved with commit 39195c4
